impress-org/givewp
# Feature: Add conditional to company name field

This PR is necessary to resolve the major issue of https://github.com/impress-org/give-gift-aid/issues/168.

## Description

The current PR implements a new field in the Personal Info section that controls the visibility of the Company field whenever it's not required. Users will now be able to clearly see when the field is not required.

## Affects

Company field on donation forms.

## Visuals

![CleanShot 2023-01-23 at 11 06 07](https://user-images.githubusercontent.com/3921017/214059471-9c4e6ad6-d265-490f-a68b-3dd43722ec87.gif)
![CleanShot 2023-01-23 at 11 06 45](https://user-images.githubusercontent.com/3921017/214059561-9ec85ba8-5757-4172-8bd3-6ed2114cba5a.gif)
![CleanShot 2023-01-23 at 11 07 18](https://user-images.githubusercontent.com/3921017/214059724-275c84b1-7b13-4731-a22a-a34f40d90a5a.gif)


## Testing Instructions

1. Make sure the Company field is optional on GiveWP Settings > Default Options
2. Open a donation form
3. See the radio list
4. Switch to Yes
5. The company field is shown
6. Change form settings or global settings to make the Company field required
7. Open the donation form
8. The radio list should not be visible but only the Company field

## Pre-review Checklist



-   [] Acceptance criteria satisfied and marked in related issue
